### ISSUE TITLE
How to set up on Centos 7?

### ISSUE THREAD

**Issue by jinqiao0423** [intel/Igvtg-kernel#11]
Hi, I am trying to set up KVMGT on Centos 7 system. But every time I try to boot the new kernel it will hang there forever with the message "A start job is running for dev_disk_by\x2uuid ....."
I downloaded the code and have done everything according to iGVT-g_Setup_Guide.txt except adding two lines to /etc/initramfs-tools/modules because there is no such file in my Centos system. I am not sure whether it has anything to do with kernel booting. Could you help me with that? Thanks.

**Comment by asadziach** [comment]
Hi,

There is also an ISO release this time around. https://01.org/igvt-g/blogs/czhou/2016/intel-gvt-g-iso-public-release-q12016?langredirect=1 you can try that to check, If your hardware is compatible? if yes then analyze how things are configured in a working system? In my experience, not everything is covered in the documentation and having access to a working system is very useful.

Regards,
Asad

**Comment by jinqiao0423** [comment]
I tried to install that .ISO by DVD but failed. The error I encountered is as follows:
(initramfs) unable to find a live medium containing a live file system. Wondering whether the reason is the too old ubuntu version.

**Comment by asadziach** [comment]
Are you able to boot? You don't need to install Ubuntu, XENGT/KVMGT will work directly off the live image.
You can use USB disk to boot and save a DVD.

**Comment by jinqiao0423** [comment]
No, even after selecting trying KVMGT without installing same problem occured though:(

**Comment by asadziach** [comment]
It might be the case that you need to edit Gvt-g kernel build configuration to suit your hardware. The kernel configuration file that comes with Gvt-g is the minimum required for VGT. I had to manually enable WiFi and SD Card drivers for my laptop.

**Comment by l1viathan** [comment]
On 05/09/2016 05:22 PM, invader36 wrote:

> Hi, I am trying to set up KVMGT on Centos 7 system. But every time I try to boot the new kernel it will hang there forever with the message "A start job is running for dev_disk_by\x2uuid ....."
> I downloaded the code and have done everything according to iGVT-g_Setup_Guide.txt except adding two lines to /etc/initramfs-tools/modules because there is no such file in my Centos system. I am not sure whether it has anything to do with kernel booting. Could you help me with that? Thanks.

I would recommend you to use your centos config:

```
# cd linux-vgt

//use your centos config
# cp /boot/config-xxx  .config
# make oldconfig

//also select CONFIG_XENGT, CONFIG_KVMGT, CONFIG_I915_VGT
# make menuconfig
```

PS, if you want to contain kvm.ko and xengt.ko in a system with
dracut, add a line to /etc/dracut.conf:

```
add_drivers+="kvm xengt"
```

HTH.

## 

Thanks,
Jike

**Comment by jinqiao0423** [comment]
I am sorry, but where is linux-vgt? THX

**Comment by asadziach** [comment]
This should be the directory where you cloned https://github.com/01org/Igvtg-kernel.git.

**Comment by jinqiao0423** [comment]
Thx a lot! Besides, I don't know where is CONFIG_XENGT, CONFIG_KVMGT, CONFIG_I915_VGT in menuconfig...

**Comment by jinqiao0423** [comment]
I see... I saw CONFIG_XENGT, CONFIG_KVMGT, CONFIG_I915_VGT in .config and they have already been set to y. I have already added
add_drivers+="kvm xengt"
to /etc/dracut.conf
But still the problem exists...

**Comment by asadziach** [comment]
Does it still hang at "A start job is running for dev_disk_by\x2uuid ....."?

**Comment by jinqiao0423** [comment]
Yes, still the same information...
I wonder whether I need to change codes inserted into grub.cfg.
I have tried changing --class ubuntu to --class centos but still the same problem..

**Comment by asadziach** [comment]
You need to understand what are the Gvt-g specific grub options and add them to your working grub configuration. The similar is true for the kernel build configuration, you should be able to build and boot a kernel on your hardware apart from Gvt-g, then add the Gvt-g specific options in there.
Hope it helps.

**Comment by jinqiao0423** [comment]
Thx! By the way, is Guest Setup steps same for Linux and Windows system?

**Comment by asadziach** [comment]
Its similar but not the same, on Linux guest you will need to copy Gvt-g kernel and modules and edit the grub file before you can set vgt=1 in Xen's .hvm file. i915.preliminary_hw_support=1 on grub line, may be required depending on guest kernel version and GPU family. 
For windows you need to install Intel's native GPU driver.

**Comment by jinqiao0423** [comment]
Hmm, not quite understand. So why and how can we install a Windows graphic driver on Linux system?

**Comment by asadziach** [comment]
the Intel graphics driver needs to be installed 'inside' windows guest.

**Comment by jinqiao0423** [comment]
Thx! If I want to create a Windows Guest, should I use command "qemu-img"? Besides, I don't see where to add .iso file information. I remember that when I used KVM, I will use "qemu-system-x86_64 -cdrom XXXXX.iso" to do so, but in the setup guide I fail to see similar words.

**Comment by zhoucc1985** [comment]
How to set up on CentOS 7.2?
I can boot the new kernel on CentOS 7.1 when I add "add_drivers+="kvm xengt"" to  /etc/dracut.conf.
But after I update the system to CentOS 7.2,the new kernel doesn't boot.
why?
Can you help me?
